v2/cmd/coil/main.go

```diff
@@ -45,7 +46,12 @@ func cmdAdd(args *skel.CmdArgs) error {
 		return convertError(err)
 	}
 
-	result, err := current.NewResult(resp.Result)
+	result_040, err := types040.NewResult(resp.Result)
```

Comment:
Thanks for reviewing !

Indeed, our final goal is to upgrade to version 1.0.0.
In order to achieve that goal, I have divided the modification into two phases.
	1. Upgrade the code base to 1.0.0 and keep the version of coil itself at 0.4.0.
	2. Upgrade the version of the coil itself to 1.0.0.
The reason for splitting the fix into two phases is that both are destructive fixes, and therefore easier to bisect if they cause problems in the future.
This PR implements the first step, so it contains some transient code like ```result_040```.
The transient codes will be removed when we upgrade coils themselves to 1.0.0.

How do you think ?

Reply:
As far as I know, adding 1.0.0 support doesn't break anything. Coil will work the same as before for 0.4 or 0.3.1 users because Coil will accept inputs from 0.4 or 0.3.1 and return the exact same as before. So transient period is not necessary, in my opinion. Am I missing something?